specification.md

```diff
@@ -6186,9 +6186,32 @@ You can define extensions in your workflow definition using its top-level `exten
 For more information about this property, see the `extensions` property in the 
 [Workflow Definition Structure section](#Workflow-Definition-Structure).
 
-Even tho users can define their own extensions, it is encouraged to use the ones provided by the specification.
-We also encourage users to contribute their extensions to the specification. That way they can be shared
-with the rest of the community.
+Besides extensions provided by the specification, users can define their custom extensions, as well as overwrite 
+the specification-provided ones.
+Note that custom extensions must define with a `namespace` value different from the Serverless Workflow
+extensions namespace ("io.serverlessworkflow.extensions"). 
+
+Defining a namespace allows runtimes to determine if they can support a defined extension or not. It also allows
+custom extension to provide their own implementation of an extension that is already provided by the specification. 
+
+For example, if you wanted to overwrite the specification-provided kpi extension, you could have:
+```json
+{
+  "extensions": [
+    {
+      "name": "My KPI extension",
+      "extensionId": "workflow-kpi-extension",
+      "namespace": "my.extensions.namespace",
+      "path": "file://myextensions/mykpi.yml"
+    }
+  ]
+}
+```
+
+Extension lookup can be performed by either specifying a URI to the extensions definition (JSON/YAML) by
+specifying it as the value of the `path` parameter.
+If the path parameter is not specified, the lookup should be done by `namespace` and `extensionId` properties.
+In this case, runtimes must provide means to map `namespace` and `extensionId` to an extension resource.
```

Comment:
Just wondering if we should not stick to just `extensionId` based lookup. That would make it way easier to figure out what is used and why. 

That would allow as well simple scheme for ids of extensions that could make it more portable and maybe even easier to share across runtimes. What do you think?

Reply:
Fine with me
@globalflea can you please add your input here as well? Your issue explicitly mentions namespace. Would the simplified lookup @mswiderski is proposing here work for you as well?

Reply:
If we stick with `extensionId` instead what's the purpose of this PR then? Because one can use the id as the namespace like `io.serverlessworkflow.extensions.kpi`

Reply:
I think that's what we are trying to figure out and need @globalflea input to see what his requirements are for this and how to best incorporate them. Please provide more info.

Reply:
Happy Holidays,
It depends on the intent of ExtensionId.  AFAIK, it functions like a pointer to the location of the extension element - i.e. it points to a specific extension within the file referred by Path parameter.  If this is the case, then it should not be used as for namespaces. 

On the other hand, if it not intended as a 'pointer', then by the same argument, i would say even either the Name or the ExtensionId is redundant - e.g Name can be of the  form "X.Y.ExtensionId.Name" or some sort of agreed convention like that

Reply:
thanks for the update. will think about it some more and update pr soon

Reply:
still looking into this...didnt forget :)